upstream/upstream.go

```diff
@@ -1047,6 +1047,41 @@ func (u *Upstream) detectFeatures(ctx context.Context) error {
 					u,
 				)
 			}
+		} else if common.IsChainIdChecksEnabled(ctx) {
```

Comment:
This is bit of a hack... It's better if we just have a separate isolated path for Validate() command.

It can be a simple loop that calls Upstream.EvmGetChainId(ctx) for them and checks. We can keep this core app logic clean and it assumes if a node gets here it's already validated.